packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -647,6 +647,38 @@ export function walk(target: IStateTreeNode, processor: (item: IStateTreeNode) =
     processor(node.storedValue)
 }
 
+export interface IReflected {
+    name: string
+    properties: { [K: string]: IType<any, any> }
+    actions: string[]
+    views: string[]
+    volatile: string[]
+}
+/**
+ * Returns a reflection of the node
+ *
+ * @export
+ * @param {IStateTreeNode} target
+ * @returns {IReflected}
+ */
+export function reflect(target: IStateTreeNode): IReflected {
+    // check all arguments
+    if (process.env.NODE_ENV !== "production") {
+        if (!isStateTreeNode(target))
+            fail("expected first argument to be a mobx-state-tree node, got " + target + " instead")
+    }
+    const node: any = getStateTreeNode(target)
+    const type = node.type.subType || node.type
+    const reflected: IReflected = {
+        name: type.name,
+        properties: type.properties,
+        actions: Object.keys(node.type.actions || {}),
+        volatile: Object.keys(node.type.volatile || {}),
+        views: Object.keys(node.type.views || {})
```

Comment:
for the views, should we somehow make clear whether a view is a function or a property? Or separately expose as `viewProperties` and `viewFunctions`? (implementation wise you could check whether the property descriptor has `get` )